### ISSUE TITLE
Rewire bridges with multiple non-contiguous edges

### ISSUE THREAD

**Issue by mdlincoln** [dSHARP-CMU/bridgesofPittsburgh#29]
For bridges with multiple non-contiguous edges, we want to create new single edges with synthetic terminal nodes:

![img_20181004_150131](https://user-images.githubusercontent.com/4392604/46496929-3aa40280-c7e7-11e8-8b8a-5a48030cacf6.jpg)

Need to implement this across all bridges.

**Comment by mdlincoln** [comment]
- [ ] Which bridge `Relations` actually _need_ rewiring? (Find by checking each `Relation` subgraph for connectedness)
- [ ] For each `Relation` to be connected, find terminal points
    - This will get very messy if bridges don't have two clear sets of terminal points... must figure out
- [ ] Create synthetic centroid poitns
- [ ] Wire new edge between synthetic points, and wire each point to its set of real target points

**Comment by mdlincoln** [comment]
Confirmed that we can identify find terminal Nodes of any given Way based on their ordering in the raw data before it's transformed into a graph. Can reference that side table during graph rewiring.

**Comment by mdlincoln** [comment]
OK, so I've worked out a first pass of this. Here's an exmaple that demonstrates what's working, and what's not.

Here is some of the original data, with all OSM `Ways` tagged as bridges highlighted in red:

<img width="883" alt="screen shot 2018-10-12 at 3 31 24 pm" src="https://user-images.githubusercontent.com/4392604/46890275-3fdefe00-ce34-11e8-9586-79744a6e5883.png">

The two bridges I've circled - Veterans Bridge and the Roberto Clemente bridge - are the ones that actually have OSM `Relations` uniting multiple `Ways` as one physical bridge. Here's the post-rewiring, which condenses each `Relation`-defined bridge into one single edge with new endpoints calculated form the centroids of its original termini (chosen using k-means):

<img width="786" alt="screen shot 2018-10-12 at 3 31 01 pm" src="https://user-images.githubusercontent.com/4392604/46890322-6f8e0600-ce34-11e8-9ca9-40228351d7c1.png">

It works (in, of course, a soft of spatially inaccurate way) for the `Relation` bridges. However, you'll note that big cluster of "bridges" from the overpass on 28 aren't united as one physical element, even though they are all marked as bridges. The highways in particular are going to play hell with us, and so I'll need to see if there are any other patterns of OSM tags that we can use to filter those out from the list of "required" bridges.
